std/uuid.d

```diff
@@ -1085,8 +1088,14 @@ unittest
     return u;
 }
 
+private @trusted
+ref T unsafeTrustingArrayElementReinterpret(T)(ref ubyte[16] arr, size_t i)
+{
+    return *cast(T*)(arr.ptr+i);
```

Comment:
Isn't this an abuse of `@trusted` based on the recent discussions and Walter's guidelines? I can easily call this function with `i==1000` and it will happily dereference an invalid pointer for me. It does not provide a proper `@safe` API.